Source/AwsCommonRuntimeKit/sdkutils/CRTAWSProfileCollection.swift

```diff
@@ -24,34 +26,39 @@ public class CRTAWSProfileCollection {
         self.rawValue = profilePointer
     }
 
-    public init(fromBuffer buffer: ByteBuffer,
-                source: CRTAWSProfileSourceType,
-                allocator: Allocator = defaultAllocator) {
+    /// Create a new profile collection by parsing text in a buffer. Primarily for testing.
+    init(fromBuffer buffer: ByteBuffer,
+         source: CRTAWSProfileSourceType,
+         allocator: Allocator = defaultAllocator) {
         var byteArray = buffer.toByteArray()
         let byteCount = byteArray.count
-        let byteBuf = byteArray.withUnsafeMutableBufferPointer { pointer -> aws_byte_buf in
+        var byteBuf = byteArray.withUnsafeMutableBufferPointer { pointer -> aws_byte_buf in
             let byteBuf = aws_byte_buf(len: byteCount,
                                        buffer: pointer.baseAddress,
                                        capacity: byteCount,
                                        allocator: allocator.rawValue)
             return byteBuf
         }
-        let pointer: UnsafePointer<aws_byte_buf> = fromPointer(ptr: byteBuf)
         self.rawValue = aws_profile_collection_new_from_buffer(allocator.rawValue,
-                                                               pointer,
+                                                               &byteBuf,
                                                                source.rawValue)
 
     }
 
-    public init(configProfileCollection: CRTAWSProfileCollection,
-                credentialProfileCollection: CRTAWSProfileCollection,
-                source: CRTAWSProfileSourceType,
-                allocator: Allocator = defaultAllocator) {
-        self.rawValue = aws_profile_collection_new_from_merge(allocator.rawValue,
-                                                              configProfileCollection.rawValue,
-                                                              credentialProfileCollection.rawValue)
+    /// Create a new profile collection by merging a config-file-based profile
+    /// collection and a credentials-file-based profile collection
+    public init?(configProfileCollection: CRTAWSProfileCollection,
```

Comment:
for all these init functions, it's debatable whether they should be `init?(...)` or `init(...) throws`

throws seems a bit safer... since the error might not be something dead-obvious like "file doesn't exist". It might be a parsing error that the user wouldn't realize unless we told them

Reply:
I agree throw is better than init?. The C code just returns null, either we should introduce proper error handling in C or just throw sdkutils_general error for now.